spec/enumerables_spec.rb

```diff
@@ -0,0 +1,174 @@
+# frozen_string_literal: true
+
+require './lib/enumerable.rb'
+
+describe Enumerable do
+  let(:arr) { [1, 2, 7, 8, 5] }
+
+  describe '#my_each should return' do
+    it 'an array holding all elements of the given array' do
+      expect(arr.my_each).to eql([1, 2, 7, 8, 5])
+    end
+
+    it 'an empty array if an empty array is given' do
+      expect([].my_each).to eql([])
+    end
+
+    it 'an array of the yield for each element if given a block' do
+      result = arr.my_each { |x| x + 2 }
+      expect(result).to eql([3, 4, 9, 10, 7])
```

Comment:
**Tip: 💭** 😄 
This is fine for your custom but actually, I strongly, strongly suggest you look at the original metho...

You would notice that it actually returns self when the block is given, i.e original array passed, so a test like following would fail:
```ruby
result = arr.my_each { |x| x + 2 }
result2 = arr.each { |x| x + 2 }
expect(result).to eql(result2)
```
What you can do to test something like this if you so choose to edit your original my_each to return self instead of the result of each. (Remember each is meant to really loop so the block should have all you need... if you needed to store result in an array you could implement that in the block

What yo ucan do to test for this is something like following
```ruby
describe 'my_each' do
    it 'calls the given block once for each element in self' do
      my_each_output = ''
      block = proc { |num| my_each_output += num.to_s }
      array.each(&block)
      each_output = my_each_output.dup
      my_each_output = ''
      array.my_each(&block)
      expect(my_each_output).to eq(each_output)
    end
  end
```

Instead of what it returns.


Hope this comment was valuable to you :)